### PR TITLE
feature/LDP-712_Convert-Nested-Objects-into-Fragments

### PR DESCRIPTION
**Pull request by roddolf**
- [LDP-712](https://jira.base22.com/browse/LDP-712) - Convert any nested object into a fragment when saving the document.
 - Add private `document._normalize()` method that convert nested objects and remove unreferenced BlankNodes.
 - Add `document.removeNamedFragment()` method that remove the specified NamedFragment.

####Breaking Changes
- Change `document.removeFragment()` method to the private `document._removeFragment()` method.

### CODE REVIEW

File: src/Auth/TokenAuthenticator.ts [hunk 1]
```diff
@@ -101,7 +102,8 @@ export class Class implements Authenticator<UsernameAndPasswordToken> {
 
 			let digestedSchema:ObjectSchema.DigestedObjectSchema = this.context.documents.getSchemaFor( expandedToken );
 
-			this.context.documents.jsonldConverter.compact( expandedToken, token, digestedSchema, this.context.documents );
+			let jsonldConverter:JSONLDConverter = new JSONLDConverter( this.context.getObjectSchema() );
```
Comment (MiguelAraCo) [inline src/Auth/TokenAuthenticator.ts]: What was the reason to change the way instances of JSONLDConverter are obtained?

**Comment by roddolf** [comment]
@MiguelAraCo 
The change was due to the necessity of the GeneralSchema in the JSONLDConverter.
And for not overload of parameters inside the JSONLDConverter, it's more clean to provide this schema in the constructor.

Or we could leve the getter `context.documents.jsonldConverter` of the documents instance, but every time the JSONLDConverter is asked for, instantiate a new one ( because the schema could be changed ).

**Comment by MiguelAraCo** [comment]
@roddolf The intention of having a singleton of `jsonldConverter` in each `Documents` module was to make them configurable (add `literalParsers`, etc.). By initializing a new one each time we would lose that flexibility (unless we hold the configuration inside of documents and initialize `jsonldConverter` with it). 

I understand the point but for now let's add an argument for the `generalSchema` in the `compact` function. Let's also add a TODO to improve the pattern being used with it.

**Comment by MiguelAraCo** [comment]
Blocked because of the critical architectural changes LDP-534 introduces

**Pull request merged**